platform-modules/batch-models/src/main/scala/org/ekstep/analytics/model/DialcodeUsageSummaryModel.scala

```diff
@@ -0,0 +1,64 @@
+package org.ekstep.analytics.model
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.ekstep.analytics.framework._
+import org.ekstep.analytics.framework.conf.AppConf
+import org.ekstep.analytics.framework.util.CommonUtil
+
+case class DialcodeScanMetrics(dialCode: String, channel: String, firstScan: Long, lastScan: Long, count: Int) extends AlgoOutput
+
+object DialcodeUsageSummaryModel extends IBatchModelTemplate[V3Event, V3Event, DialcodeScanMetrics, MeasuredEvent] with Serializable {
+
+    implicit val className = "org.ekstep.analytics.model.DialcodeUsageSummaryModel"
+
+    override def name: String = "DialcodeUsageSummaryModel"
+
+    override def preProcess(data: RDD[V3Event], config: Map[String, AnyRef])(implicit sc: SparkContext): RDD[V3Event] = {
+        data
+    }
+
+    override def algorithm(input: RDD[V3Event], config: Map[String, AnyRef])(implicit sc: SparkContext): RDD[DialcodeScanMetrics] = {
+        val mapDialCodes = (x: RDD[V3Event]) => x.map(y => {
+            var dailcode: List[String] = List()
```

Comment:
There is a typo in the dialcode variable name. Use mutable.ListBuffer instead of a mutable variable.